### PR TITLE
Fix transmission loader tests

### PR DESCRIPTION
**Pull request by bmagyar**
follow-up of #633 so it sits on top of those changes currently, I'm not 100% convinced about the change in the transmission_parser, please change my mind

### CODE REVIEW

File: hardware_interface/include/hardware_interface/hardware_info.hpp [hunk 1]
```diff
@@ -87,7 +87,7 @@ struct ActuatorInfo
   double offset = 0.0;
 };
 
-/// Contains semantic info about a given transmission loaded from XML (URDF)
+/// Contains semantic info about a given transmission loaded from URDF for a transmission
```
Comment (destogl) [inline hardware_interface/include/hardware_interface/hardware_info.hpp:90]: It's kind a double...

```suggestion
/// Contains semantic info about a given transmission loaded from URDF
```

File: transmission_interface/CMakeLists.txt [hunk 2]
```diff
@@ -75,6 +75,7 @@ if(BUILD_TESTING)
     test/simple_transmission_loader_test.cpp
   )
   target_include_directories(test_simple_transmission_loader PUBLIC include hardware_interface)
+  target_link_libraries(test_simple_transmission_loader gmock_main)
```
Comment (destogl) [inline transmission_interface/CMakeLists.txt:78]: is this really needed? It doesn't look familiar. Shouldn't we link using `ament_target_dependencies` and than `ament_cmake_gmock` or something like that?

**Review by destogl — CHANGES_REQUESTED** [review]
Just a two small comments.

You wanted to remove the Exceptions. Do you whan to do this in follow-up? Or open a `good-first-issue`?

File: transmission_interface/CMakeLists.txt (hunk 2 above)
Comment (bmagyar) [reply transmission_interface/CMakeLists.txt:78]: this is how you can avoid defining a main for your test files

**Review by destogl — COMMENTED** [review]
Just one small nitpick for keeping the code clean

File: transmission_interface/CMakeLists.txt (hunk 2 above)
Comment (bmagyar) [reply transmission_interface/CMakeLists.txt]: removed, everythning seem to work still

**Review by destogl — APPROVED** [review]

**Pull request merged**